cocos/renderer/CCTextureCache.cpp

```diff
@@ -238,7 +242,9 @@ void TextureCache::unbindImageAsync(const std::string& filename)
     if (_imageInfoQueue && !_imageInfoQueue->empty())
     {
         std::string fullpath = FileUtils::getInstance()->fullPathForFilename(filename);
-        auto found = std::find_if(_imageInfoQueue->begin(), _imageInfoQueue->end(), [&fullpath](ImageInfo* ptr)->bool{ return ptr->asyncStruct->filename == fullpath; });
+        //BPC PATCH - shared_ptr
+        auto found = std::find_if(_imageInfoQueue->begin(), _imageInfoQueue->end(), [&fullpath](std::shared_ptr<ImageInfo> ptr)->bool{ return ptr->asyncStruct->filename == fullpath; });
```

Comment:
```suggestion
        auto found = std::find_if(_imageInfoQueue->begin(), _imageInfoQueue->end(), [&fullpath](std::shared_ptr<ImageInfo> const &ptr)->bool{ return ptr->asyncStruct->filename == fullpath; });
```
If you wanna be consistent with cocos stuff, you might want to avoid that copy.

Reply:
Probably negligible in the long run though.